pkg/apis/kops/v1alpha2/cluster.go

```diff
@@ -349,12 +349,21 @@ type ClusterSubnetSpec struct {
 	// ProviderID is the cloud provider id for the objects associated with the zone (the subnet on AWS)
 	ProviderID string `json:"id,omitempty"`
 
-	// Egress defines the method of traffic egress for this subnet
-	Egress string `json:"egress,omitempty"`
+	// Egress is used to add more routes of precreated NAT gateways or NAT instances or VPC peering connections
+	Egress []EgressSpec `json:"egress,omitempty"`
 
+	// Type define which one if the internal types (public, utility, private) the network is
 	Type SubnetType `json:"type,omitempty"`
 }
 
+// EgressSpec defines routes of precreated NAT gateways or NAT instances or VPC peering connections
+type EgressSpec struct {
+	CIDR                 string `json:"cidr,omitempty"`
+	NatGateway           string `json:"natGateway,omitempty"`
```

Comment:
So one of the ideas of `egress` being a string was that only one of NatGateway, Instance or VpcPeeringConnection can be set, and we can tell them apart by their prefix.  And this also avoids tieing the kops model to AWS specifics, but lets us be a little more cloud independent.